azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java

```diff
@@ -1255,4 +1259,119 @@ public void accept(MessageSender a) {
+     * returns null to indicate timeout.
+     * 
+     * @param partitionId  Partition to get information about. Must be one of the partition ids returned by getRuntimeInformation.
+     * @return CompletableFuture which returns an EventHubPartitionRuntimeInformation on success, or null on timeout.  
+     * @throws Exception On failures other than timeout.
+     */
+    public CompletableFuture<EventHubPartitionRuntimeInformation> getPartitionRuntimeInformation(String partitionId) {
+    	Map<String, String> request = new HashMap<String, String>();
+        request.put(ClientConstants.MANAGEMENT_ENTITY_TYPE_KEY, ClientConstants.MANAGEMENT_PARTITION_ENTITY_TYPE);
+        request.put(ClientConstants.MANAGEMENT_ENTITY_NAME_KEY, this.eventHubName);
+        request.put(ClientConstants.MANAGEMENT_PARTITION_NAME_KEY, partitionId);
+        request.put(ClientConstants.MANAGEMENT_OPERATION_KEY, ClientConstants.READ_OPERATION_VALUE);
+
+        // The current implementation is secretly synchronous. Retain the async method signature for parity with
+        // dotnet and because a more sophisticated reimplementation may really be async.
+        CompletableFuture<EventHubPartitionRuntimeInformation> retval = new CompletableFuture<EventHubPartitionRuntimeInformation>();
+        
+        try
+        {
+        	Map<String, Object> rawdata = managementWithRetry(request);
+    		retval.complete(new EventHubPartitionRuntimeInformation(
+				(String)rawdata.get(ClientConstants.MANAGEMENT_ENTITY_NAME_KEY),
+				(String)rawdata.get(ClientConstants.MANAGEMENT_PARTITION_NAME_KEY),
+				(long)rawdata.get(ClientConstants.MANAGEMENT_RESULT_BEGIN_SEQUENCE_NUMBER),
+				(long)rawdata.get(ClientConstants.MANAGEMENT_RESULT_LAST_ENQUEUED_SEQUENCE_NUMBER),
+				(String)rawdata.get(ClientConstants.MANAGEMENT_RESULT_LAST_ENQUEUED_OFFSET),
+				((Date)rawdata.get(ClientConstants.MANAGEMENT_RESULT_LAST_ENQUEUED_TIME_UTC)).toInstant()));
+        }
+        catch (Throwable t)
+        {
+        	retval.completeExceptionally(t);
+        }
+		
+		return retval;
+    }
+    
+    private Map<String, Object> managementWithRetry(Map<String, String> request) throws Throwable
+    {
+        Map<String, Object> rawdata = null;
+        Instant endTime = Instant.now().plus(this.underlyingFactory.getOperationTimeout());
+        Throwable lastException = null;
+        
+        while ((rawdata == null) && Instant.now().isBefore(endTime)) {
+	        ManagementChannel mgmt = this.underlyingFactory.getManagementChannel();
+	        try {
+				rawdata = mgmt.doManagementRequestResponse(this.underlyingFactory.getReactorScheduler(), request).get();
+			}
+	        catch (InterruptedException | ExecutionException e) {
+	        	// RequestResponse channel does not have any retry built into it, it always shuts down
+	        	// on any kind of error. So call close just to be tidy, then do a wait and try again.
```

Comment:
>on any kind of error [](start = 13, length = 20)

Does this mean - none of the errors are retriable &- we need not retry at all ? Lets check .net client behavior...